tests/beta/test_default_tensor.py

```diff
@@ -331,6 +338,304 @@ def test_hermitian_exceptions(self):
+
+        dev = tensornet_device_2_wires
+
+        observable = np.array([[0, 1], [1, 0]])
+        obs_nodeA = tensornetwork.Node(observable, name="ObsA")
+        obs_nodeB = tensornetwork.Node(observable, name="ObsB")
+        dev._add_edge(obs_nodeA, 1, dev._nodes["state"][0], 0)
+        dev._add_edge(obs_nodeB, 1, dev._nodes["state"][1], 0)
+
+        # check_connected is designed to raise an exception if nodes are not connected
+        # test passes if no exception is raised
+        tensornetwork.check_connected([dev._nodes["state"][0], obs_nodeA])
+        tensornetwork.check_connected([dev._nodes["state"][1], obs_nodeB])
+
+
+    def test_create_nodes_from_tensors(self, tensornet_device_2_wires):
+        """Tests that the create_nodes_from_tensors method adds nodes to the tensor
+        network properly."""
+
+        dev = tensornet_device_2_wires
+
+        assert len(dev._nodes["state"]) == 2
+        A = np.array([[0, 1], [1, 0]])
+        new_node = dev.create_nodes_from_tensors([A], [[0]], ["GateA"], key="state")
+        assert new_node[0] in dev._nodes["state"]
+        assert len(dev._nodes["state"]) == 3
+
+        new_nodes = dev.create_nodes_from_tensors([A, A], [[0], [1]], ["GateA", "GateB"], key="state")
+        assert all([node in dev._nodes["state"] for node in new_nodes])
+
+        obs_nodes = dev.create_nodes_from_tensors([A, A], [[0], [1]], ["ObsA", "ObsB"], key="observables")
+        assert all(node in dev._nodes["observables"] for node in obs_nodes)
+
+
+    @pytest.mark.parametrize("method", ["auto", "greedy", "branch", "optimal"])
+    def test_contract_to_ket_correct(self, tensornet_device_3_wires, method, tol):
+        """Tests that the _contract_to_ket method contracts down to a single node with the correct tensor."""
+
+        dev = tensornet_device_3_wires
+
+        dev.apply("PauliX", [0], [])
+        dev.apply("Hadamard", [1], [])
+
+        assert "contracted_state" not in dev._nodes
+        dev._contract_to_ket(method)
+        assert "contracted_state" in dev._nodes
+        cont_state = dev._nodes["contracted_state"]
+        dev._contract_to_ket(method)  # should not change anything
+        assert dev._nodes["contracted_state"] == cont_state
+
+        expected = np.outer(np.outer([0., 1.], [1 / np.sqrt(2), 1 / np.sqrt(2)]), [1., 0.]).reshape([2,2,2])
+
+        assert np.allclose(cont_state.tensor, expected, atol=tol, rtol=0)
+        assert cont_state.name == "Ket"
+
+    @pytest.mark.parametrize("method", ["auto", "greedy", "branch", "optimal"])
+    def test_state(self, tensornet_device_3_wires, method, tol):
+        """Tests that the _state method produces the correct state after contraction."""
+
+        dev = tensornet_device_3_wires
+
+        dev.apply("PauliX", [0], [])
+        dev.apply("Hadamard", [1], [])
+
+        assert "contracted_state" not in dev._nodes
+        ket = dev._state()
+        assert "contracted_state" in dev._nodes
```

Comment:
Based on this, somehow it might be worth including a note, that simply obtaining `dev._state()` makes a lot of difference with this device. Maybe a note in the docstring saying that a contraction takes place?

Reply:
Or another solution could be renaming it (e.g. get_contracted_state() (?)): to me `dev._state` seems like something that would not trigger any major logic based on how other devices work, but rather just something that helps with keeping track of the state at times.

Reply:
Goal here is to give users access to the state without giving them unnecessary technical details about what is going on under the hood (which, as you can see, is somewhat different from how other plugins work). So I prefer a similar naming scheme to what we might see in other plugins.

I've added a bit more info to the docstring though.